### PR TITLE
Add hostname parameter for TCP probe

### PR DESCRIPTION
**Pull request by lspiehler**
This adds the ability to set the TLS server name for TCP probes using the hostname parameter just like #823 did for HTTP probes. It was mentioned by @Nothing4You in #624, and its something I need for monitoring in my environment as well.

### CODE REVIEW

File: prober/handler.go [hunk 1]
```diff
@@ -107,6 +107,14 @@ func Handler(w http.ResponseWriter, r *http.Request, c *config.Config, logger lo
 		}
 	}
 
+	if module.Prober == "tcp" && hostname != "" {
```
[the remark this answers is not in the document]
Comment (lspiehler) [reply prober/handler.go:110]: Test has been created. Thanks!

**Review by electron0zero — APPROVED** [review]
lgtm :+1:

**Comment by electron0zero** [comment]
looking for comments from @mem @roidelapluie, will merge if there are no comments :)

**Comment by gitanuj** [comment]
This seems like a missing feature for tcp probes (since http probes already expose hostname). Would love to see this ship soon!

File: prober/handler.go [hunk 2]
```diff
@@ -137,6 +145,15 @@ func Handler(w http.ResponseWriter, r *http.Request, c *config.Config, logger lo
 	h.ServeHTTP(w, r)
 }
 
+func setTLSServerName(hostname string, module *config.Module) error {
```
Comment (electron0zero) [inline prober/handler.go:148]: error is always nil? so doesn't make sense to return error in this case.

also, can we not inline this assignment?

Comment (lspiehler) [reply prober/handler.go]: I removed the function and made the assignment inline in d207f64. Thanks!

**Review by electron0zero — APPROVED** [review]

**Comment by electron0zero** [comment]
merged, thanks @lspiehler :)

**Pull request merged**